src/sagas/wallet.js

```diff
@@ -596,11 +596,29 @@ export function* onWalletReloadData() {
   }
 }
 
+export function* onResetWallet() {
+  const wallet = yield select((state) => state.wallet);
+
+  if (wallet) {
+    // wallet.stop() will remove all event listeners and call
+    // hathorLib.wallet.cleanWallet
+    wallet.stop({ cleanStorage: true });
+
+    return;
+  }
+
+  // Wallet was not initialized yet, this might happen if resetWallet
+  // is called from the PinScreen. There is no event listeners to cleanup
+  // so we can call the cleanWallet method directly.
+  walletUtil.cleanWallet();
```

Comment:
This method unsubscribe addresses from the ws and other actions from the connection. Doesn't it throw errors when you call it without a `wallet` object? We shouldn't have a connection yet.

Reply:
That's a valid point, currently, it is not throwing an error because it's defaulting to the old `WebSocketHandler`, which is an `EventEmitter`. I will refactor this to only clean the storage

Reply:
Done in 52289657b65b452f4b1f999bc11a7057b048b928